src/schema/root.js

```diff
@@ -382,6 +394,39 @@ export const resolvers = {
 
       return { deleted: !res.error, error: res.error };
     },
+    async renewLoan(parent, args, context, info) {
+      // NOTE FOR FURTHER DEVELOPMENT
+      // In the long run, we would like to be able to retrieve the loans status from a. o. cicero to see if the loan can be renewed.
+      // This way, we can check if the loan can be renewed without calling the renewLoan function, which would be a better user experience.
+
+      const { loanId, agencyId, dryRun = false } = args;
+
+      const userId = getUserId({ agencyId, context });
+
+      if (!userId) {
+        return {
+          renewed: false,
+          error: "User or agency not found",
+        };
+      }
```

Comment:
Forslag: Kunne måske godt have en conditional return for hver af de 3.  (`loanId`, `agencyId`, `userId`). Hvis `agencyId` eller `loanId` mangler, vil det også være fint at returnere før vi kalder `getUserId`?

Reply:
det er ret defensive, det er jo efterhånden programmører der kalder denne endpoint. jeg wrapper agencyid og loanid/orderid i en early-return, så er vi på den sikre side og programmøren kan finde ud af, hvis det var agencyID eller loan/orderId han/hun manglede.

Reply:
det er ret defensive, det er jo efterhånden programmører der kalder denne endpoint. jeg wrapper agencyid og loanid/orderid i en early-return, så er vi på den sikre side og programmøren kan finde ud af, hvis det var agencyID eller loan/orderId han/hun manglede.